src/app/pages/BorrowPage/components/AddCollateralModal/AddCollateralModal.tsx

```diff
@@ -0,0 +1,169 @@
+import { ReviewStep } from './ReviewStep';
+import { fromWei } from 'utils/blockchain/math-helpers';
+import { Asset } from 'types';
+
+type AddCollateralModalProps = {
+  loan: ActiveLoan;
+  onCloseModal: () => void;
+  onSubmit: (
+    collateralToken: Asset,
+    loanId: string,
+    depositAmount: string,
+  ) => void;
+  liquidationPrice?: React.ReactNode;
+  positionSize?: string;
+};
+
+enum Step {
+  FORM,
+  REVIEW,
+}
+
+export const AddCollateralModal: React.FC<AddCollateralModalProps> = ({
+  loan: item,
+  onSubmit,
+}) => {
+  const canInteract = useCanInteract();
+  const tokenDetails = useMemo(
+    () =>
+      AssetsDictionary.getByTokenContractAddress(item?.collateralToken || ''),
+    [item?.collateralToken],
+  );
+  const loanToken = useMemo(
+    () => AssetsDictionary.getByTokenContractAddress(item?.loanToken || ''),
+    [item?.loanToken],
+  );
+  const [amount, setAmount] = useState('');
+  const [step, setStep] = useState(Step.FORM);
+  const { value: balance } = useAssetBalanceOf(tokenDetails.asset);
+  const weiAmount = useWeiAmount(amount);
+
+  const { send, ...tx } = useApproveAndAddMargin(
+    tokenDetails.asset,
+    item.loanId,
+    weiAmount,
+  );
+  const { checkMaintenance, States } = useMaintenance();
+  const topupLocked = checkMaintenance(States.ADD_TO_MARGIN_TRADES);
+
+  const valid = useIsAmountWithinLimits(weiAmount, '1', balance);
+
+  const {
+    value: principalAsCollateral,
+    loading: loadingRate,
+  } = useDenominateAssetAmount(
+    loanToken.asset,
+    tokenDetails.asset,
+    item.principal,
+  );
+
+  const currentCollateralRatio = useMemo(
+    () => calculateCollateralRatio(principalAsCollateral, item.collateral),
+    [item.collateral, principalAsCollateral],
+  );
+
+  const maintenanceRatio = useMemo(
+    () => bignumber(fromWei(item.maintenanceMargin)).add(100).toFixed(0),
+    [item.maintenanceMargin],
+  );
+
+  const currentLiquidationPrice = useMemo(
+    () =>
+      calculateLiquidationPrice(
+        item.principal,
+        item.collateral,
+        maintenanceRatio,
+      ),
+    [item.collateral, item.principal, maintenanceRatio],
+  );
+
+  const newCollateralAmount = useMemo(
+    () => bignumber(item.collateral).add(weiAmount).toFixed(0),
+    [item.collateral, weiAmount],
+  );
+
+  const newLiquidationPrice = useMemo(
+    () =>
+      calculateLiquidationPrice(
+        item.principal,
+        bignumber(item.collateral).add(weiAmount).toString(),
+        maintenanceRatio,
+      ),
+    [item.collateral, item.principal, maintenanceRatio, weiAmount],
+  );
+
+  const newCollateralRatio = useMemo(
+    () =>
+      calculateCollateralRatio(
+        principalAsCollateral,
+        bignumber(item.collateral).add(weiAmount).toString(),
+      ),
+    [item.collateral, principalAsCollateral, weiAmount],
+  );
+
+  const handleNextStep = useCallback(() => setStep(Step.REVIEW), []);
+
+  const canSubmit = useMemo(
```

Comment:
This is totally misleading. One would expect that if `canSubmit` is `true` that you actually can submit a tx. But it's the exact opposite and in `CollateralForm` and `ReviewStep` it's used to determine if submit buttons are disabled.

Rename it to something more appropriate (e.g. `isSubmitDisabled`) and also rename `canSubmit` properties on both `CollateralForm` and `ReviewStep` to `disabled`.